### ISSUE TITLE
Illegal Instruction (core dumped)

### ISSUE THREAD

**Issue by in3rsha** [mimblewimble/grin-miner#146]
I'm trying to mine on my server, but the `grin-miner` exits and returns this message just as it is about to get going:

    Illegal Instruction (core dumped)

So basically the miner starts, but it seems to crash with the above error around the time it receives its first job.

### Steps

I have been running the following to start the miner:

1. `grin server run`
2. `grin wallet listen`
3. `grin-miner`

This worked on my local computers, but I am seeing this error on my server (even though I have followed the exact same steps).

### Server Specs

```
OS: Ubuntu 16.04.5 LTS x86_64 
Host: X8DTU 1234567890 
Kernel: 4.4.0-141-generic 
CPU: Intel Xeon X5675 (24) @ 3.068GHz 
GPU: Matrox Electronics Systems Ltd. MGA G200eW WPCM450 
Memory: 29887MiB / 128931MiB
```

```
$ rustc --version
rustc 1.31.1 (b6c32da9b 2018-12-18)
```

### Logs

```
Jan 17 16:55:35.195 INFO This is Grin-Miner version 1.0.2 (git v1.0.2), built for x86_64-unknown-linux-gnu by rustc 1.31.1 (b6c32da9b 2018-12-18).
Jan 17 16:55:35.196 DEBG Built with profile "debug", features "CURSIVE, DEFAULT, TUI" on Thu, 17 Jan 2019 11:20:30 GMT.
Jan 17 16:55:35.196 DEBG Starting solvers
Jan 17 16:55:35.196 DEBG No mining plugin dir provided by config. Using default plugin dir: "/home/admini/grin-miner/target/debug/plugins"
Jan 17 16:55:35.196 DEBG Loading miner plugin: /home/admini/grin-miner/target/debug/plugins/cuckaroo_cpu_avx2_29.cuckooplugin
Jan 17 16:55:35.200 DEBG Loading miner plugin: /home/admini/grin-miner/target/debug/plugins/cuckaroo_cpu_avx2_29.cuckooplugin
Jan 17 16:55:36.201 WARN Connection Status: Connected to Grin server at 127.0.0.1:3416.
Jan 17 16:55:36.211 DEBG sending request: {"id":"0","jsonrpc":"2.0","method":"getjobtemplate","params":null}
Jan 17 16:55:37.008 DEBG Received message: {"id":"Stratum","jsonrpc":"2.0","method":"job","params":{"difficulty":1,"height":2717,"job_id":0,"pre_pow":"00010000000000000a9d000000005c40b38800000e7ed12058b1ee660259251fca8ac506b891d5b8a6d0aa28433bacb443cd86ddd6e179fdd0f11f59494e784d0fa45a36c9f5a146777e52a549fb4fec1791faef778504f03dbccae9a67c352c2675630ebe85aadaa8124b0a5d0bc829b5d84dd0dc87cc44f136ef0e48c4a54e9f1a22cd4533a388946c0d250dbbfb3961f75afcb3bfb3150b4c52da2ce981ccb5f48687b487c5e009d1fe8a39aed5c6e47a1ccc19fbea530607e68e52a4a08c7dd767599c8e5b61cc6316fb14dec9f65d9b0000000000002e9a0000000000002015000001738b60bd0800000398"}}

Jan 17 16:55:37.009 DEBG Received request type: job
Jan 17 16:55:37.009 INFO Got a new job: JobTemplate { height: 2717, job_id: 0, difficulty: 1, pre_pow: "00010000000000000a9d000000005c40b38800000e7ed12058b1ee660259251fca8ac506b891d5b8a6d0aa28433bacb443cd86ddd6e179fdd0f11f59494e784d0fa45a36c9f5a146777e52a549fb4fec1791faef778504f03dbccae9a67c352c2675630ebe85aadaa8124b0a5d0bc829b5d84dd0dc87cc44f136ef0e48c4a54e9f1a22cd4533a388946c0d250dbbfb3961f75afcb3bfb3150b4c52da2ce981ccb5f48687b487c5e009d1fe8a39aed5c6e47a1ccc19fbea530607e68e52a4a08c7dd767599c8e5b61cc6316fb14dec9f65d9b0000000000002e9a0000000000002015000001738b60bd0800000398" }
Jan 17 16:55:37.009 DEBG Received message: {"id":"0","jsonrpc":"2.0","method":"getjobtemplate","result":{"difficulty":1,"height":2717,"job_id":0,"pre_pow":"00010000000000000a9d000000005c40b38800000e7ed12058b1ee660259251fca8ac506b891d5b8a6d0aa28433bacb443cd86ddd6e179fdd0f11f59494e784d0fa45a36c9f5a146777e52a549fb4fec1791faef778504f03dbccae9a67c352c2675630ebe85aadaa8124b0a5d0bc829b5d84dd0dc87cc44f136ef0e48c4a54e9f1a22cd4533a388946c0d250dbbfb3961f75afcb3bfb3150b4c52da2ce981ccb5f48687b487c5e009d1fe8a39aed5c6e47a1ccc19fbea530607e68e52a4a08c7dd767599c8e5b61cc6316fb14dec9f65d9b0000000000002e9a0000000000002015000001738b60bd0800000398"},"error":null}

Jan 17 16:55:37.009 DEBG Received response with id: 0
Jan 17 16:55:37.009 INFO Got a job at height 2717 and difficulty 1
Jan 17 16:55:37.102 DEBG Miner received message: ReceivedJob(2717, 0, 1, "00010000000000000a9d000000005c40b38800000e7ed12058b1ee660259251fca8ac506b891d5b8a6d0aa28433bacb443cd86ddd6e179fdd0f11f59494e784d0fa45a36c9f5a146777e52a549fb4fec1791faef778504f03dbccae9a67c352c2675630ebe85aadaa8124b0a5d0bc829b5d84dd0dc87cc44f136ef0e48c4a54e9f1a22cd4533a388946c0d250dbbfb3961f75afcb3bfb3150b4c52da2ce981ccb5f48687b487c5e009d1fe8a39aed5c6e47a1ccc19fbea530607e68e52a4a08c7dd767599c8e5b61cc6316fb14dec9f65d9b0000000000002e9a0000000000002015000001738b60bd0800000398")
Jan 17 16:55:37.103 DEBG Pause message sent
Jan 17 16:55:37.103 DEBG Resume message sent
Jan 17 16:55:37.103 DEBG Miner received message: ReceivedJob(2717, 0, 1, "00010000000000000a9d000000005c40b38800000e7ed12058b1ee660259251fca8ac506b891d5b8a6d0aa28433bacb443cd86ddd6e179fdd0f11f59494e784d0fa45a36c9f5a146777e52a549fb4fec1791faef778504f03dbccae9a67c352c2675630ebe85aadaa8124b0a5d0bc829b5d84dd0dc87cc44f136ef0e48c4a54e9f1a22cd4533a388946c0d250dbbfb3961f75afcb3bfb3150b4c52da2ce981ccb5f48687b487c5e009d1fe8a39aed5c6e47a1ccc19fbea530607e68e52a4a08c7dd767599c8e5b61cc6316fb14dec9f65d9b0000000000002e9a0000000000002015000001738b60bd0800000398")
Jan 17 16:55:37.172 DEBG solver_thread - solver_loop_rx got msg: Pause
Jan 17 16:55:37.172 DEBG solver_thread - solver_loop_rx got msg: Resume
```

### Notes

I have tried to solve the problem using the solutions posted here, but it hasn't worked for me: https://github.com/mimblewimble/grin-miner/issues/37

### Question

Does anyone know why the `grin-miner` is not working on this hardware?

**Comment by in3rsha** [comment]
I seem to have fixed the problem by changing the **Miner Plugin Configuration** in `grin-miner.toml`.

The configuration was set to this originally:

```
[[mining.miner_plugin_config]]
plugin_name = "cuckaroo_cpu_avx2_29"
[mining.miner_plugin_config.parameters]
nthreads = 4
```

And those settings seemed to be causing the problem. So I just changed the settings to ones that were working on my local computer:

```
[[mining.miner_plugin_config]]
plugin_name = "cuckaroo_cpu_compat_29"
[mining.miner_plugin_config.parameters]
nthreads = 1
```

I'm guessing the settings may have been configured in `grin-miner.toml` based on my computer's hardware when I compiled the program.

I have changed the settings to something that works. However, I am completely new to using `grin-miner`, so they're probably less than optimal. But at least it gives me something to work with for now.
